### PR TITLE
Improve redis performance by switching to hashes

### PR DESCRIPTION
**Pull request by karthikv**
We currently use the redis keys function with the wildcard (*) selector to access a list of projects, screens, etc. Yet the documentation for redis directly says that "Warning: consider KEYS as a command that should only be used in production environments with extreme care. It may ruin performance when it is executed against large databases ... Don't use KEYS in your regular application code" (see http://redis.io/commands/keys).

As a result, I suggest we switch to using redis hashes. Having each key in a hash represent the id of a project, screen, etc. and the value represent the data associated that id will be a simple way to map over our models while maintaining much higher efficiency.

I've been working on this and will attach a commit as soon as issue #6 is resolved, as my code is dependent on those changes.

### CODE REVIEW

**Comment by karthikv** [comment]
@seanmonstar Just added my commit to this issue.

File: lib/crud.js [hunk 1]
```diff
@@ -1,88 +1,72 @@
...
-        db.get(objectItem, function(errObject, objectItem) {
-          objectItem = JSON.parse(objectItem);
-          objectList.unshift(objectItem);
-
-          if (counter === objects.length - 1) {
-            callback(null, objectList);
-          }
-        });
+    } else {
+      objectList = _.map(objectList, function(object) {
```
Comment (seanmonstar) [inline lib/crud.js]: `objectList` is originally an object, or array? If it's an array, you don't need underscore to use `map`.

File: lib/crud.js [hunk 2]
```diff
@@ -116,13 +105,20 @@ exports.update = function(req, keyName, allowedFields, db, callback) {
...
-  callback = callback || utils.placeholder;
-  db.del(keyName, function(err) {
-    callback(err);
+exports.remove = function(key, id, db, callback) {
+  callback = callback || utils.noop;
+  db.hdel(key, id, function(err, numRemoved) {
+    if (err) {
+      callback(err);
+    } else if (numRemoved != 1) {
+      callback('Asked to delete one object, but ' + numRemoved +
```
Comment (seanmonstar) [inline lib/crud.js]: pass an Error instance, not a String.

**Comment by karthikv** [comment]
@seanmonstar Added a commit that addresses the issues you mentioned.

**Comment by seanmonstar** [comment]
r+

**Pull request merged**